spring-session/src/main/java/org/springframework/session/hazelcast/config/annotation/web/http/EnableHazelcastHttpSession.java

```diff
@@ -71,4 +72,13 @@
 	 */
 	String sessionMapName() default "spring:session:sessions";
 
+	/**
+	 * This is the flag that will be used to persist sessions after
+	 * something has changed or immediately. By default, session will be persisted
+	 * immediately.
+	 *
+	 * @return the flag to persist sessions after changes.
+	 */
+	boolean persistOnSessionChange() default false;
```

Comment:
I find this javadoc misleading. In my mind, this flag is not about when session will be saved (after changes or immediately). It's about avoiding save when changes are not detected. But we must state that changes might not be detected (as @rwinch pointed out). Also, we should state that flag can help to reduce the possibility of race conditions.